cdap-support-bundle/bin/collect_support_bundle.py

```diff
@@ -0,0 +1,655 @@
+    "-l",
+    help="CDAP pipeline name",
+    type=str
+)
+parser.add_argument(
+    "--run-id",
+    "-r",
+    help="CDAP Run id",
+    type=str
+)
+parser.add_argument(
+    "--runtime-namespace",
+    "-t",
+    help="CDAP Runtime namespace. Namespace where pipeline is deployed",
+    type=str
+)
+def main(namespace, username, password, cdap_ns, pipeline_name, run_id, runtime_namespace):
+
+    fetch_instance_command = f'kubectl get cdapmasters.cdap.cdap.io --no-headers -o custom-columns=":metadata.name" -n {namespace}'
+    instance_name = subprocess.getoutput(fetch_instance_command)
+    if(instance_name == "" or instance_name is None):
+        print(f"Error : Unable to fetch the instance name. Please check for namespace in input parameters, either it's invalid or no instance is available in {namespace}")
+        exit()
+
+    auth_header = ""
+    if username is not None and password is not None:
+        passToken = base64.b64encode(bytes(f'{username}:{password}', 'utf-8'))
+        passToken = passToken.decode('utf-8')
+        output = subprocess.getoutput(f'kubectl -n {namespace} exec -t cdap-{instance_name}-appfabric-0 -c appfabric -- bash -c \'curl -s cdap-{instance_name}-external-authentication:10009/token -H "Authorization:Basic {passToken}"\'')
+        try:
+            str_obj = json.loads(output)
+            auth_header = f"-H \"Authorization: Bearer {str_obj['access_token']}\""
+        except JSONDecodeError as e:
+            print(e)
+            print('Error : Unable to authenticate due to Invalid credentials. Please check for username and password in input parameters')
+            exit()
+    else:
+        output_check_auth = subprocess.getoutput(
+            f'kubectl exec -t cdap-{instance_name}-runtime-0 -n {namespace} -c runtime -- bash -c \'curl -s cdap-{instance_name}-router:11015/v3/system/services/status {auth_header}\'')
+        try:
+            str_obj_check_auth = json.loads(output_check_auth)
+            if "appfabric" in str_obj_check_auth:
+                print("Authentication Disabled. Log extraction process started.")
+            else:
+                print('Error : Authentication is enabled for the kubernetes namespace provided, please also provide username and password parameter while running the script')
+                exit()
+
+        except JSONDecodeError as e:
+            print(f"message : {e}")
+            print('Error : Authentication is enabled for the kubernetes namespace provided, please also provide username and password parameter while running the script')
+            exit()
+
+
+    update_progress(1)
+    current_datetime = datetime.now()
+
+    current_date_time = current_datetime.strftime("%m-%d-%YT%H%M%S")
+    directoryName = "cdap-logs-" + current_date_time
+
+    parent_dir = os.getcwd() + '/'
+    path = os.path.join(parent_dir, directoryName)
+
+    os.mkdir(path)
+    update_progress(4)
+    config_file = "config_file"
+    pod_logs = "pod_logs"
```

Comment:
drop the _logs suffix for all the directory names except for directories that actually contain pod logs (e.g: cdap_services_logs contains logs for cdap system service pods, so let's keep the dir name as is.)

Reply:
done